services/QuillLMS/app/lib/long_process_notifier.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class LongProcessNotifier < ApplicationService
+  attr_reader :error, :threshold, :options
+
+  def initialize(error, threshold, options = {})
+    @error = error
+    @threshold = threshold
+    @options = options
+  end
+
+  def run
```

Comment:
Note, since you are using `attr_reader`, you should use the non-instance variables in the class, e.g. `threshold` not `@threshold`